packages/ckeditor5-engine/tests/model/schema.js

```diff
@@ -2411,6 +2411,173 @@ describe( 'Schema', () => {
 			// } );
 		} );
 
+		describe( 'allowChildren', () => {
+			it( 'allows item in another item', () => {
+				schema.register( 'paragraph' );
+
+				schema.register( '$root', {
+					allowChildren: 'paragraph'
+				} );
+
+				expect( schema.checkChild( root1, r1p1 ) ).to.be.true;
+			} );
+
+			it( 'supports the array syntax', () => {
+				schema.register( 'paragraph' );
+				schema.register( 'blockQuote' );
+
+				schema.register( '$root', {
+					allowChildren: [ 'paragraph', 'blockQuote' ]
+				} );
+
+				expect( schema.checkChild( root1, r1p1 ) ).to.be.true;
+				expect( schema.checkChild( root1, r1bQ ) ).to.be.true;
+			} );
+
+			it( 'supports circular references', () => {
+				schema.register( '$root', {
+					allowChildren: [ 'paragraph', 'blockQuote' ]
+				} );
+
+				schema.register( 'paragraph', {
+					allowChildren: 'blockQuote'
+				} );
+
+				schema.register( 'blockQuote', {
+					allowChildren: 'paragraph'
+				} );
+
+				expect( schema.checkChild( r1p1, r1bQ ) ).to.be.true;
+				expect( schema.checkChild( r1bQ, r1p1 ) ).to.be.true;
+			} );
+
+			it( 'supports self-reference', () => {
+				schema.register( '$root', {
+					allowChildren: 'paragraph'
+				} );
+
+				schema.register( 'paragraph', {
+					allowChildren: 'paragraph'
+				} );
+
+				expect( schema.checkChild( r1p1, r1p1 ) ).to.be.true;
+			} );
+
+			it( 'passes $root>$paragraph>div>blockQuote - deep nesting', () => {
+				schema.register( '$root', {
+					allowChildren: 'paragraph'
+				} );
+
+				schema.register( 'paragraph', {
+					allowChildren: 'div'
+				} );
+
+				schema.register( 'div', {
+					allowChildren: 'blockQuote'
+				} );
+
+				schema.register( 'blockQuote' );
+
+				const paragraph = new Element( 'paragraph' );
+				root1._appendChild( paragraph );
+
+				const div = new Element( 'div' );
+				paragraph._appendChild( div );
+
+				const blockQuote = new Element( 'blockQuote' );
+				div._appendChild( blockQuote );
+
+				expect( schema.checkChild( root1, paragraph ) ).to.be.true;
+				expect( schema.checkChild( paragraph, div ) ).to.be.true;
+				expect( schema.checkChild( div, blockQuote ) ).to.be.true;
+
+				expect( schema.checkChild( paragraph, blockQuote ) ).to.be.false;
+				expect( schema.checkChild( div, paragraph ) ).to.be.false;
+			} );
+
+			it( 'should remove allowChildren from definition', () => {
+				schema.register( '$root', {
+					allowChildren: 'paragraph'
+				} );
+
+				schema.register( 'paragraph' );
+
+				expect( schema.getDefinition( '$root' ) ).to.deep.equal( {
+					allowAttributes: [],
+					allowIn: [],
+					name: '$root'
+				} );
+			} );
```

Comment:
Maybe it could be useful to be able to see what elements might be children for some element?

Reply:
Seeing all allowed children for the specific definition looks very useful for _at least_ debugging, even if we don't have any real case scenario for now 👍🏻